lib/puppet/provider/foreman_smartproxy/rest_v3.rb

```diff
@@ -4,7 +4,7 @@
   def proxy
     @proxy ||= begin
       r = request(:get, 'api/v2/smart_proxies', :search => %{name="#{resource[:name]}"})
-      raise Puppet::Error.new("Proxy #{resource[:name]} cannot be retrieved: #{error_message(r)}") unless success?(r)
+      raise Puppet::Error.new("Proxy #{resource[:name]} cannot be retrieved from Foreman: #{error_message(r)}") unless success?(r)
```

Comment:
would it make sense to point to production.log on the foreman for more info? I'm assuming the error message in most cases here might not be very informative regarding why it failed other than http status code

Reply:
Yes, most often it was HTTP 5xx which people didn't understand. That can be in Foreman or Apache or in between. On HTTP 4xx we can display an error:

https://github.com/theforeman/puppet-foreman/blob/746229243abb98995981f90d1a777e096d201108/lib/puppet/provider/foreman_resource/rest_v3.rb#L97-L99

Sadly, I think we don't often have this.